### ISSUE TITLE
SIGABRT on Android

### ISSUE THREAD

**Issue by sebastianpfluegelmeier** [Kitt-AI/snowboy#234]
When i make a new SnowboyDetect object i get this Error from the Log and the App crashes:

`01-01 16:00:38.264 25080-25080/com.example.pfluegelmeiersebast.testserver A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 25080 (bast.testserver)`
                                                                                  
`                                                                                  [ 01-01 16:00:38.264  2263: 2263 W/         ]
                                                                                  debuggerd: handling request: pid=25080 uid=10145 gid=10145 tid=25080`

I am using a Samsung Galaxy Tab A with Android 7.0 installed

**Comment by sebastianpfluegelmeier** [comment]
I also checked all Permissions and i also get a stack trace
```
01-02 06:29:45.714 23207-23207/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-02 06:29:45.714 23207-23207/? A/DEBUG: Build fingerprint: 'samsung/gtaxlltexx/gtaxllte:7.0/NRD90M/T585XXU2BQE4:user/release-keys'
01-02 06:29:45.715 23207-23207/? A/DEBUG: Revision: '6'
01-02 06:29:45.715 23207-23207/? A/DEBUG: ABI: 'arm'
01-02 06:29:45.715 23207-23207/? A/DEBUG: pid: 23179, tid: 23179, name: bast.testserver  >>> com.example.pfluegelmeiersebast.testserver <<<
01-02 06:29:45.715 23207-23207/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
01-02 06:29:45.715 23207-23207/? A/DEBUG:     r0 00000000  r1 00005a8b  r2 00000006  r3 00000008
01-02 06:29:45.715 23207-23207/? A/DEBUG:     r4 f5e5258c  r5 00000006  r6 f5e52534  r7 0000010c
01-02 06:29:45.715 23207-23207/? A/DEBUG:     r8 e7896ca0  r9 e986ded8  sl 00000000  fp e9876154
01-02 06:29:45.715 23207-23207/? A/DEBUG:     ip 00000054  sp ffc84448  lr f46d74c7  pc f46d9d30  cpsr 600f0010
01-02 06:29:45.718 23207-23207/? A/DEBUG: backtrace:
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #00 pc 0004ad30  /system/lib/libc.so (tgkill+12)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #01 pc 000484c3  /system/lib/libc.so (pthread_kill+34)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #02 pc 0001dd99  /system/lib/libc.so (raise+10)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #03 pc 00019521  /system/lib/libc.so (__libc_android_abort+34)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #04 pc 00017160  /system/lib/libc.so (abort+4)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #05 pc 000beb3b  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so (_ZN9__gnu_cxx27__verbose_terminate_handlerEv+226)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #06 pc 000882b5  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so (_ZN10__cxxabiv111__terminateEPFvvE+4)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #07 pc 0008792f  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #08 pc 000880c9  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #09 pc 000c9a40  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #10 pc 000c9c2c  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so (__gnu_Unwind_RaiseException+148)
01-02 06:29:45.718 23207-23207/? A/DEBUG:     #11 pc 000ca748  /data/app/com.example.pfluegelmeiersebast.testserver-2/lib/arm/libsnowboy-detect-android.so (_Unwind_RaiseException+20)
```


this are the permissions i checked:


        boolean permission =
                ContextCompat.checkSelfPermission(this,
                        Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED
                &&
                ContextCompat.checkSelfPermission(this,
                        Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED
                &&
                ContextCompat.checkSelfPermission(this,
                        Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED
                &&
                ContextCompat.checkSelfPermission(this,
                        Manifest.permission.ACCESS_WIFI_STATE) == PackageManager.PERMISSION_GRANTED
                &&
                ContextCompat.checkSelfPermission(this,
                        Manifest.permission.INTERNET) == PackageManager.PERMISSION_GRANTED;

**Comment by stmax82** [comment]
I get the same error (SIGABRT with code -6) in the Android Demo-App:

```
01-02 08:23:05.655 10389-10389/ai.kitt.snowboy.demo A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 10389 (tt.snowboy.demo)
                                                            
                                                            [ 01-02 08:23:05.656  2259: 2259 W/         ]
                                                            debuggerd: handling request: pid=10389 uid=10148 gid=10148 tid=10389
```

It happens at "new SnowboyDetector(...)" in RecordingThread.java:

```
    private SnowboyDetect detector = new SnowboyDetect(commonRes, activeModel);
```

I tried manually opening and reading the files "commonRes" and "activeModel". They can both be read successfully, so there doesn't seem to be a permission problem there.

Is there a way to get useful error messages or a stacktrace?

**Comment by stmax82** [comment]
For my case I have found and fixed the problem:

When you git clone the repo on a Windows system the files in the assets directory exist but are corrupted. I suppose it has something to do with symlinks. (Similar problem with the jniLibs directory...)

I replaced the files under assets and the jniLibs directory and I'm since able to build the demo app on Windows.

The question remains though - Is there a way to get more useful error messages or a stacktrace?

Also I'm wondering why the .so files and swig wrappers aren't included and have to be manually built? It'd make building and using snowboy much easier - it would even work out of the box on Windows.
